astrovascpy/bloodflow.py

```diff
@@ -684,10 +686,18 @@ def _solve_linear(laplacian, input_flow):
                         laplacian += factor * sparse.eye(np.shape(laplacian)[0])
                         result = linalg.spsolve(laplacian, input_flow)
 
-    if os.getenv("BACKEND_SOLVER_BFS", "petsc") == "scipy":
+    if os.getenv("BACKEND_SOLVER_BFS", "scipy") == "scipy":
+        if bool(int(os.getenv("DEBUG_BFS", "0"))) and (MPI_RANK == 0):
+            scipy_res_norm = np.linalg.norm(input_flow - laplacian * result)
+            PETSc.Sys.Print(f"-> SciPy residual norm = {scipy_res_norm}")
         return result
 
     # PETSc-related part!
+    if MPI_RANK == 0 and n_nodes < 2e6:
+        L.warning(
```

Comment:
same remark as above. I would define the log message a constant to prevent code duplication